### DISCUSSION TITLE
Fixing one parameter per batch of experiments

### DISCUSSION THREAD

**Discussion by iandoxsee** [meta-pytorch/botorch#2243] [Q&A]
Hi all, I have a use case where one feature needs to be fixed among all generated experiments within a single batch. 

For example, 5 experiments are generated per batch and x1 can be chosen from the range 0-100, but all 5 experiments must be executed at the same x1. In subsequent batches, a new x1 value can be used but again must be shared among all 5 experiments. 

I found a reference for "process-constrained batch Bayesian optimization" and I'm wondering if this is supported with any of the existing constraints features within Botorch. Thank you for your help!

https://dl.acm.org/doi/pdf/10.5555/3294996.3295100

**Comment by Balandat** [comment]
Yeah the basic operations in that paper (i.e. fixing the value of a particular input) is supported using the `fixed_features` input to `optimize_acqf`: https://github.com/pytorch/botorch/blob/e68efb5d9bc20c5fab280f4a00558ee9e6dc5fea/botorch/optim/optimize.py#L429

Alternatively you could also use `FixedFeatureAcqusitionFunction`: https://github.com/pytorch/botorch/blob/e68efb5d9bc20c5fab280f4a00558ee9e6dc5fea/botorch/acquisition/fixed_feature.py#L52

You can use this to assemble the  `pc-BO(basic)` algorithm from the paper after first optimizing the unconstrained acquisition function and then fixing the value for subsequent elements of the batch (just be sure to set `X_pending` with the previous points on the new acquisition function objects to properly account for the previous points in the batch). 

However, for the `pc-BO(nested)` algorithm you would need to know what the proposed `alpha_c^{GP-UCB}` function is. I don't think this is defined anywhere in the paper... at least I haven't found it.
